### PR TITLE
Fix #1840 : Introduce central utility file for append text in testing

### LINKED ISSUE #1840: Introduce central utility file for appendText in Testing
**Is your feature request related to a problem? Please describe.**
We are using `appendText` function in the test files extensively.

**Describe the solution you'd like**
So we can create a central file for this as a helper method. 

```
/**
   * Appends the specified text to a view. This is needed because Robolectric doesn't seem to
   * properly input digits for text views using 'android:digits'. See
   * https://github.com/robolectric/robolectric/issues/5110 for specifics.
   */
  private fun appendText(text: String): ViewAction {
    val typeTextViewAction = typeText(text)
    return object : ViewAction {
      override fun getDescription(): String = typeTextViewAction.description

      override fun getConstraints(): Matcher<View> = typeTextViewAction.constraints

      override fun perform(uiController: UiController?, view: View?) {
        // Appending text only works on Robolectric, whereas Espresso needs to use typeText().
        if (Build.FINGERPRINT.contains("robolectric", ignoreCase = true)) {
          (view as? EditText)?.append(text)
          testCoroutineDispatchers.runCurrent()
        } else {
          typeTextViewAction.perform(uiController, view)
        }
      }
    }
  }
```

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes #1840 
```appendText()``` is moved to a central utility file.
All use cases have been updated.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by prayutsu** [comment]
@anandwana001 I'll fix the lint issue after the review

**Comment by anandwana001** [comment]
> @anandwana001 I'll fix the lint issue after the review

These lint issues are minor, I can review the PR, but cannot approve till everything looks good because sometimes fixing one thing may cause other issue so to be perfectly looks everything good needs everything before PR review.

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.hamcrest.CoreMatchers
+import org.hamcrest.Matcher
+
+/**
+ * Appends the specified text to a view. This is needed because Robolectric doesn't seem to
+ * properly input digits for text views using 'android:digits'. See
+ * https://github.com/robolectric/robolectric/issues/5110 for specifics.
+ */
+
+fun appendText(text: String, testCoroutineDispatchers: TestCoroutineDispatchers): ViewAction {
```
Comment (anandwana001) [inline testing/src/main/java/org/oppia/testing/TestingUtils.kt]: How about if we create a particular class and keep this method inside that, and there we can also do constructor injection of `testCoroutineDispatchers`. Also, run the test cases in both espresso and robolectric to see if this works after your change, just to confirm things work fine. 
Take the example of `ProfileTestHelper`. 
In this way, if in future we get more such methods which are repetitive in nature, we can put them here.

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    override fun getConstraints(): Matcher<View> {
+      return CoreMatchers.allOf(ViewMatchers.isEnabled())
+    }
+
+    override fun perform(uiController: UiController?, view: View?) {
+      (view as? EditText)?.append(text)
+      testCoroutineDispatchers.runCurrent()
+    }
+  }
+}
```
Comment (anandwana001) [inline testing/src/main/java/org/oppia/testing/TestingUtils.kt]: add a single empty line at the end of the file.

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt (hunk 1 above)
Comment (anandwana001) [inline testing/src/main/java/org/oppia/testing/TestingUtils.kt]: How about we create a particular class here and do the constructor injection for `testCoroutineDispatchers`. Also, try to run the test cases with your changes to see if test cases work fine in both the espresso as well as robolectric. 
Take the example of `ProfileTestHelper`. 
This way I think if more such repetitive methods occur in future which is more inclined towards in use of all the test cases, we can put them here.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @prayutsu 
Added a few comments.

**Comment by prayutsu** [comment]
> > @anandwana001 I'll fix the lint issue after the review
> 
> These lint issues are minor, I can review the PR, but cannot approve till everything looks good because sometimes fixing one thing may cause other issue so to be perfectly looks everything good needs everything before PR review.

Sure

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt (hunk 2 above)
Comment (prayutsu) [reply testing/src/main/java/org/oppia/testing/TestingUtils.kt]: yup that's actually the lint error

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt (hunk 1 above)
Comment (prayutsu) [reply testing/src/main/java/org/oppia/testing/TestingUtils.kt]: Okay, that is surely a good idea.

**Comment by prayutsu** [comment]
@anandwana001 PTAL

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import android.widget.EditText
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import org.hamcrest.CoreMatchers
+import org.hamcrest.Matcher
+import javax.inject.Inject
+
+/**
+ * Appends the specified text to a view. This is needed because Robolectric doesn't seem to
```
Comment (anandwana001) [inline testing/src/main/java/org/oppia/testing/TestingUtils.kt]: This comment is about `appendText` function, not the class itself. 
We have to put some class level comment which helps us understand hat this class do and shift this comment to `appendText` function.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @prayutsu added a nit change

File: testing/src/main/java/org/oppia/testing/TestingUtils.kt (hunk 3 above)
Comment (prayutsu) [reply testing/src/main/java/org/oppia/testing/TestingUtils.kt]: Okay, so should I shift this comment above the function and add another comment for the class?

Comment (anandwana001) [reply testing/src/main/java/org/oppia/testing/TestingUtils.kt]: yes

Comment (anandwana001) [reply testing/src/main/java/org/oppia/testing/TestingUtils.kt]: as per our discussion, we are going for single file `EditTextInputAction`

**Comment by prayutsu** [comment]
@anandwana001 Ran all the tests on robolectric, all the tests are passing. PTAL

**Comment by prayutsu** [comment]
@anandwana001 I looked at the failing check, but couldn't understand it. Can you please help in this?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
LGTM, Thanks @prayutsu this seems really helpful and a clean way. 
Please update it with the latest develop and check a few newly updated test class having `appendText`. 
By the time you update, I will make sure no more `appendText` arrive your way.

**Comment by prayutsu** [comment]
@anandwana001 Made changes to all the new files. PTAL

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @prayutsu

**Comment by rt4914** [comment]
@prayutsu This PR looks really great. Thanks a lot.
@anandwana001 Thanks for helping out.

**Pull request merged**